packages/react/src/index.tsx

```diff
@@ -1,47 +1,75 @@
-import React, { createContext, useContext, useEffect, useState } from 'react'
+import React, { PropsWithChildren, createContext, useContext, useEffect, useState } from 'react'
 import ButtonImg from './medium-passport-button.svg'
 
-const ParallelContext = createContext({ parallel: null })
+import { ParallelApiRecord, loadParallel } from '@parallelmarkets/vanilla'
+import type { AuthCallbackResult, Parallel } from '@parallelmarkets/vanilla'
+import { AccreditationsApiResponse } from './accreditation_api_types'
+import { ProfileApiResponse } from './profile_api_types'
+import { BlockchainApiResponse } from './blockchain_api_types'
+import { IdentityApiResponse } from './identity_api_types'
 
-export const ParallelProvider = ({ parallel, children, ...props }) => {
+export * from './accreditation_api_types'
+export * from './profile_api_types'
+export * from './blockchain_api_types'
+export * from './identity_api_types'
+
+type LoadParallelPromise = ReturnType<typeof loadParallel>
+type LoadParallelResult = Awaited<ReturnType<typeof loadParallel>>
+type ParallelContextValue = { parallel?: LoadParallelPromise }
+
+const ParallelContext = createContext<ParallelContextValue>({ parallel: undefined })
+
+type ParallelProviderProps = PropsWithChildren<{
+  parallel: LoadParallelPromise
+}>
+
+export const ParallelProvider = ({ parallel, children, ...props }: ParallelProviderProps) => {
   return (
     <ParallelContext.Provider value={{ parallel }} {...props}>
       {children}
     </ParallelContext.Provider>
   )
 }
 
-const isPromise = (thing) => {
-  return thing !== null && typeof thing === 'object' && typeof thing.then === 'function'
+const isPromise = (thing: unknown): thing is PromiseLike<unknown> => {
```

Comment:
This is a **Type Guard** with a **Type Predicate**. It's a common pattern for helper functions that makes **Type Narrowing** easier.

We're telling TypeScript that we'll do a _runtime_ check (the body of the function) to make sure this object conforms well enough for TS to trust that is the given type (the `is SomeType`—or type predicate).

Narrowing: https://www.typescriptlang.org/docs/handbook/2/narrowing.html

Using Predicates: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates